File: Kurs C++/Zad6/dodaj.cpp
//
// Created by Mateusz Pater on 08.04.2016.
//

#include "dodaj.h"


double dodaj::oblicz() {
    return a->oblicz()+b->oblicz();
}

std::string dodaj::opis() {

    return "Wartosc " + std::to_string(oblicz());

}

std::string dodaj::formatuj_wyrazenie() {
    return "(" + a->formatuj_wyrazenie() + " + " + b->formatuj_wyrazenie() + ")";
}

dodaj::dodaj(wyrazenie *a, wyrazenie *b){ this->a = a; this->b = b; }

File: Kurs Java/Zadanie 3/src/com/company/Dodaj.java
package com.company;

/**
 * author @pater
 */
public class Dodaj extends Operator2Arg {

    public Dodaj(Wyrazenie a, Wyrazenie b) {
        super(a,b);
    }

    @Override
    public double oblicz() {
        return a.oblicz() + b.oblicz();
    }
    @Override
    public String toString() {
        return "(" + a + "+" + b + ")";
    }
}
